### PR TITLE
[NEW] Translation via MS translate

### PR DESCRIPTION
**Pull request by mrsimpson**
Adds Microsoft's translation service (https://translator.microsoft.com/) as a provider for translation of messages.
In addition to implementing the interface (similar to google and DeepL), a small change has been done in order to display the translation provider on the UI.

### CODE REVIEW

File: app/autotranslate/server/settings.js [hunk 1]
```diff
@@ -20,6 +20,9 @@ Meteor.startup(function() {
 		}, {
 			key: 'deepl-translate',
 			i18nLabel: 'AutoTranslate_DeepL',
+		}, {
+			key: 'microsoft-translate',
+			i18nLabel: 'AutoTranslate_Microsoft',
```
Comment (rodrigok) [inline app/autotranslate/server/settings.js]: Missing the translation on `en.i18n.json`

File: app/autotranslate/server/settings.js [hunk 2]
```diff
@@ -54,4 +57,18 @@ Meteor.startup(function() {
 				_id: 'AutoTranslate_ServiceProvider', value: 'deepl-translate',
 			}],
 	});
+
+	settings.add('AutoTranslate_MicrosoftAPIKey', '', {
+		type: 'string',
+		group: 'Message',
+		section: 'AutoTranslate_Microsoft',
+		public: true,
+		i18nLabel: 'AutoTranslate_APIKey',
```
Comment (rodrigok) [inline app/autotranslate/server/settings.js]: Missing the translation on en.i18n.json

File: app/autotranslate/server/settings.js [hunk 3]
```diff
@@ -54,4 +57,18 @@ Meteor.startup(function() {
 				_id: 'AutoTranslate_ServiceProvider', value: 'deepl-translate',
 			}],
 	});
+
+	settings.add('AutoTranslate_MicrosoftAPIKey', '', {
+		type: 'string',
+		group: 'Message',
+		section: 'AutoTranslate_Microsoft',
+		public: true,
```
Comment (rodrigok) [inline app/autotranslate/server/settings.js]: Should not be public since it's not used on the client side

File: app/autotranslate/server/msTranslate.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+			}
+			return language;
+		});
+		const url = `${ this.apiEndPointUrl }&to=${ targetLanguages.join('&to=') }`;
+		try {
+			const result = HTTP.post(url, {
+				headers: {
+					'Ocp-Apim-Subscription-Key': this.apiKey,
+					'Content-Type': 'application/json; charset=UTF-8',
+				},
```
Comment (rodrigok) [inline app/autotranslate/server/msTranslate.js]: This part could be extracted to another method since it's duplicated on `_translateMessage`

File: app/autotranslate/server/settings.js (hunk 2 above)
Comment (mrsimpson) [reply app/autotranslate/server/settings.js]: This one? https://github.com/assistify/Rocket.Chat/blob/core/feature/ms-translate/packages/rocketchat-i18n/i18n/en.i18n.json#L446

Comment (rodrigok) [reply app/autotranslate/server/settings.js]: Yes, it's missing here

File: app/autotranslate/server/settings.js (hunk 3 above)
Comment (mrsimpson) [reply app/autotranslate/server/settings.js]: Applies to the other APIKeys too!

File: app/autotranslate/server/settings.js (hunk 2 above)
Comment (rodrigok) [reply app/autotranslate/server/settings.js]: Ah, sure, sorry.

Comment (mrsimpson) [reply app/autotranslate/server/settings.js]: Sorry, @rodrigok , I don't get it. `AutoTranslate_APIKey` is in `en.i18n.json`, see the above link. `AutoTranslate_Microsoft ` is missing, I'll add that.

File: app/autotranslate/server/settings.js (hunk 3 above)
Comment (mrsimpson) [reply app/autotranslate/server/settings.js]: Changed all API-Keys  setting in translation context to `private`

File: app/autotranslate/server/msTranslate.js (hunk 4 above)
Comment (mrsimpson) [reply app/autotranslate/server/msTranslate.js]: too true. 
Extracted to `_translate`

**Comment by mrsimpson** [comment]
@rodrigok I updated the code as per all comments.
Manually tested:
- [x] Translating messages (en -> de, de -> en)
- [x] Translating attachment descriptions
- [x] Multi-Languages in mutliple lines
- [x] Error handling when entering a wrong key

File: package-lock.json [hunk 5]
```diff
@@ -181,7 +181,7 @@
 				},
 				"minimist": {
 					"version": "1.2.0",
-					"resolved": "https://registry.npmjs.org/minimist/-/minimist-1.2.0.tgz",
+					"resolved": "http://registry.npmjs.org/minimist/-/minimist-1.2.0.tgz",
```
Comment (rodrigok) [inline package-lock.json]: Can you revert the changes on this file since it only changed the protocol of the URLs?

File: app/autotranslate/server/msTranslate.js [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+		// There are multi-sentence-messages where multiple sentences come from different languages
+		// This is a problem for translation services since the language detection fails.
+		// Thus, we'll split the message in sentences, get them translated, and join them again after translation
+		const msgs = message.msg.split('\n').map((msg) => ({ Text: msg }));
+		try {
+			translations = this._translate(msgs, targetLanguages);
+		} catch (e) {
+			logger.microsoft.error('Error translating message', e);
+		}
+		return translations;
```
Comment (rodrigok) [inline app/autotranslate/server/msTranslate.js]:
```suggestion
		// There are multi-sentence-messages where multiple sentences come from different languages
		// This is a problem for translation services since the language detection fails.
		// Thus, we'll split the message in sentences, get them translated, and join them again after translation
		const msgs = message.msg.split('\n').map((msg) => ({ Text: msg }));
		try {
			return this._translate(msgs, targetLanguages);
		} catch (e) {
			logger.microsoft.error('Error translating message', e);
		}
		return {};
```

File: app/autotranslate/server/msTranslate.js [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+	_translateAttachmentDescriptions(attachment, targetLanguages) {
+		let translations = {};
+		try {
+			translations = this._translate([{
+				Text: attachment.description || attachment.text,
+			}], targetLanguages);
+		} catch (e) {
+			logger.microsoft.error('Error translating message attachment', e);
+		}
+		return translations;
```
Comment (rodrigok) [inline app/autotranslate/server/msTranslate.js]:
```suggestion
		try {
			return this._translate([{
				Text: attachment.description || attachment.text,
			}], targetLanguages);
		} catch (e) {
			logger.microsoft.error('Error translating message attachment', e);
		}
		return {};
```

**Comment by rodrigok** [comment]
@mrsimpson almost perfect :)

File: package-lock.json (hunk 5 above)
Comment (mrsimpson) [reply package-lock.json]: Strange. I actually try to avoid committing the package-lock. Anyway: I'll revert it.

File: app/autotranslate/server/msTranslate.js (hunk 7 above)
Comment (mrsimpson) [reply app/autotranslate/server/msTranslate.js]: I am really not a fan of this code style, but well - you are the maintainer ;)

**Comment by mrsimpson** [comment]
@rodrigok I merged the latest `develop` and then checked out the `package-lock.json`.
